src/main/java/com/amazonaws/kinesisvideo/java/service/JavaKinesisVideoServiceClient.java

```diff
@@ -114,6 +114,10 @@ private static AWSCredentials createAwsCredentials(
 
         final KinesisVideoCredentials kinesisVideoCredentials = credentialsProvider.getCredentials();
 
+        if (kinesisVideoCredentials == null) {
+            return null;
```

Comment:
May want to add a log line here, in case we ever need to differentiate between this and the other return null above.

Reply:
+1